src/lib/env-vars.ts

```diff
@@ -109,42 +109,44 @@ export const SUPERVISOR_CONFIG_VAR_PROPERTIES: {
 	},
 };
 
-export const HOST_CONFIG_VAR_PROPERTIES: {
-	[k: string]: JSONSchema6Definition;
-} = {
-	RESIN_HOST_CONFIG_disable_splash: {
-		enum: ['0', '1'],
-		description: 'Enable / Disable the rainbow splash screen',
-		default: '1',
-	},
-	RESIN_HOST_CONFIG_dtparam: {
-		type: 'string',
-		description: 'Define DT parameters',
-		default: '"i2c_arm=on","spi=on","audio=on"',
-	},
-	RESIN_HOST_CONFIG_enable_uart: {
-		enum: ['0', '1'],
-		description: 'Enable / Disable UART',
-		default: '1',
+export const DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES: Array<{
+	capableDeviceTypes: Set<string>;
+	properties: Dictionary<JSONSchema6Definition>;
+}> = [
+	{
+		capableDeviceTypes: new Set([
```

Comment:
Why a `Set`?

Reply:
Thought it would make searching a bit more performant (O(1) vs O(n) on arrays).
Can go with arrays if you prefer having primitives.

Reply:
I'd say an array is fine, particularly at this size

Reply:
Alternatively you could do:
```ts
_([...]).keyBy().mapValues(() => true).value()
```
or just directly do
```ts
{
	'raspberry-pi': true,
	...
}
```
and have an object lookup

Reply:
Done :+1:

Reply:
Just to be clear, I went with the array.
Imo though a Set sounds more appropriate than `keyBy` or `{ 'raspberry-pi': true, ... }`, since the spec describes this as a native hashmap, it sounds weird to try replicate exactly that w/ a different way.